tests/keras/metrics_test.py

```diff
@@ -112,6 +113,22 @@ def test_sparse_top_k_categorical_accuracy():
 
     assert failure_result == 0
 
+    # Test correctness if the shape of y_true is (num_samples,)
+    y_pred = K.variable(np.array([[0.3, 0.2, 0.1], [0.1, 0.2, 0.7]]))
```

Comment:
Could you use pytest parametrization? I think you made a PR with it last time if I remember correctly. `@pytest.mark.parametrize`. I would allow you to avoid this code duplicate and also avoid a for loop which is not great for tests.

Reply:
In my updated version, I made ```y_pred``` and ```y_true``` parametrization, but not ```k```. This is because I want to keep the check for ```success_result```, ```partial_result``` and ```failure_result``` explicitly, they have different meanings, that will make other developer easy to understand the output of this function and test. This is different from ```backend_test``` where the test output of all parameterized inputs have the same meaning(backend function output or layer output).

Reply:
No, I don‘t think it's more reasonable.
* We can't classify as is, because there are other scenarios. For example, the TF dataset API can produce input as ```(num_samples, 1)``` and ```(num_samples,)```, the latter one is the _usually_ case(see my PR description), not the only case. And Keras can accept Tensor as well, but in you suggestion, there is no coverage.
* Here is unit test of ```sparse_top_k_categorical_accuracy```, not integrated test, so we should only test the input of this function, rather the input of whole workflow. There are only two candidate data format for this function, but Keras can accept different data formats and it not relevant to this function.

Thanks.

Reply:
You are right. It seems I was confused about the behaviour of the dataset API.

 I'll ask you to make `y_pred` a variable in the test, and not a parameter, since it does not change across test runs. Thanks!

Reply:
I don't think this change is needed, because we may add test case later for different ```y_pred```, and I think it's more clear to put ```y_pred``` and ```y_true``` together. In theory, ```y_pred``` and ```y_test``` has equal status. Thanks.

Reply:
That's well thought.